src/main/java/scribe_java/GatekeeperLogin.java

```diff
@@ -49,26 +52,40 @@ public void getGatekeeperGetAccessToken(HttpServletRequest request) {
     }
 
     public void getGatekeeperGrantAccessToken(String callback, String state, String scope) {
-        accessToken = null;
         oAuthBean.initGatekeeperService(callback, state, scope);
 
         try {
             OAuth2AccessToken inAccessToken = oAuthBean.getAberfitnessService().getAccessTokenClientCredentialsGrant();
 
             if (!(inAccessToken instanceof GatekeeperOAuth2AccessToken))
-                return;
+                throw new Exception("inAccessToken isn't instanceof GatekeeperOAuth2AccessToken");
 
-            accessToken = (GatekeeperOAuth2AccessToken) inAccessToken;
+            AuthStorage.setApplicationToken((GatekeeperOAuth2AccessToken) inAccessToken);
             return;
 
         } catch (Exception e) {
-            e.printStackTrace();
+            System.err.println("[GatekeeperLogin.getGatekeeperGrantAccessToken] Message:" + e.getMessage() + " Cause: " + e.getCause());
+        }
+    }
+
+    public boolean validateAccessToken(String accessToken) {
+        try {
+            JWTClaimsSet claimsSet = GatekeeperJsonTokenExtractor.instance().getJWTClaimSet(accessToken);
+            System.out.println("Token Issued By: " + claimsSet.getIssuer());
+
+            /*if (!claimsSet.getAudience().contains("fitbit-ingest-service")) //Disabled for testing purposes.
```

Comment:
If we are merging into development we'll probably need this to be enabled

Reply:
I would have enabled it but not sure on what the audience string will be yet as the fitbit-ingest-scope seems to have been removed from the list of supported scopes at [Openid Configuration](https://docker2.aberfitness.biz/gatekeeper/.well-known/openid-configuration). I've asked adam about it but he hasn't responded yet. It's not really that important at the minuet.